features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -218,8 +222,17 @@ bool ATHandler::get_debug() const
 
 ATHandler::~ATHandler()
 {
+    ScopedLock <ATHandler> lock(*this);
     set_file_handle(NULL);
 
+    while (_event_id > 0) {
```

Comment:
Before this there should be a 

    if (_event_id != 0 && eventqueue->cancel(_event_id)) {
        _event_id = 0;
    }

That's how this works in single-threaded case. (The cancel can only fail if the OOB is about to run in a second thread now).